test/check-nulecule

```diff
@@ -0,0 +1,110 @@
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+from testlib import *
+import json
+import os
+import sys
+import unittest
+
+@unittest.skipIf("rhel-7" == os.environ.get("TEST_OS", ""), "Skipping check-nulecule on rhel-7.")
+class KubernetesCase(MachineCase):
+
+    # HACK: https://github.com/GoogleCloudPlatform/kubernetes/issues/8311
+    # Work around for the fact that kube-apiserver doesn't notify about startup
+    # We wait until available or timeout.
+    def wait_api_server(self, port=8080, timeout=60, scheme='http'):
+        waiter = """
+        port=%d
+        timeout=%d
+        scheme=%s
+        for a in $(seq 0 $timeout); do
+            if curl -o /dev/null -k -s $scheme://localhost:$port; then
+                break
+            fi
+            sleep 0.5
+        done
+        """ % (port, timeout * 2, scheme)
+        self.machine.execute(script=waiter)
+
+    # HACK - https://bugzilla.redhat.com/show_bug.cgi?id=1243827
+    #
+    # HACK - https://github.com/GoogleCloudPlatform/kubernetes/issues/11222
+    # The default install of kubernetes does not let you start pods due to
+    # ServiceAccount admission control. So remove that.
+    def fix_apiserver_config(self):
+        self.machine.execute("sed -i /etc/kubernetes/apiserver -e 's/4001/2379/' -e 's/,ServiceAccount//'")
+
+
+@unittest.skipIf("rhel-7" == os.environ.get("TEST_OS", ""), "Skipping check-nulecule on rhel-7.")
+class TestKubernetes(KubernetesCase):
+    def setUp(self):
+        MachineCase.setUp(self)
+        m = self.machine
+
+        self.fix_apiserver_config()
+        m.execute("systemctl start etcd kube-apiserver kube-controller-manager kube-scheduler docker kube-proxy kubelet")
+
+        self.wait_api_server()
+
+@unittest.skipIf("rhel-7" == os.environ.get("TEST_OS", ""), "Skipping check-kubernetes on rhel-7.")
+class TestKubernetes(KubernetesCase):
+    def setUp(self):
+        MachineCase.setUp(self)
+        m = self.machine
+
+        self.fix_apiserver_config()
+        m.execute("systemctl start etcd kube-apiserver kube-controller-manager kube-scheduler docker kube-proxy kubelet")
+
+        m.upload(["mock-k8s-tiny-app.json"], "/tmp")
+        self.wait_api_server()
+
+    def testDashboard(self):
+        b = self.browser
+        m = self.machine
+
+        self.login_and_go("/kubernetes/cluster", host=None)
+        b.wait_in_text("#node-list", "127.0.0.1")
+
+        # Successfully deploy via dialog
+        b.click("#deploy-app")
+        b.wait_popup("deploy-app-dialog")
+        b.set_val("#deploy-app-nulecule-image", "submod/helloapache")
```

Comment:
Does this download the image during a test run?  It might be better to pre-download it into the test images so that the test doesn't fail when there is network trouble, and the test might also be faster.